JamJar/JamJarPageViewController.swift

```diff
@@ -0,0 +1,371 @@
+//
+//  JamJarPageViewController.swift
+//  JamJar
+//
+//  Created by Ethan Riback on 5/2/16.
+//  Copyright © 2016 JamJar. All rights reserved.
+//
+//  Note: This class has become a victim of code 
+//  cloning. Due to deadlines, this can't be reversed
+//  before the release of version 1.0. TODO: undo
+//  the code cloning.
+//
+
+import UIKit
+import AVKit
+import AVFoundation
+import SCLAlertView
+
+class JamJarPageViewController: BaseViewController, updateVideoDelegate, UITableViewDelegate, UITableViewDataSource, UIPickerViewDataSource, UIPickerViewDelegate {
+    
+    weak var jamjar: JamJarGraph!
+    weak var concert: Concert!
+    weak var video: Video!
+    var reasonsToFlag: [String]!
+    
+    // UI Element
+    @IBOutlet weak var jamJarContainerView: UIView!
+    @IBOutlet weak var titleLabel: UILabel!
+    @IBOutlet weak var uploaderLabel: UILabel!
+    @IBOutlet weak var viewCountLabel: UILabel!
+    @IBOutlet weak var artistsLabel: UILabel!
+    @IBOutlet weak var venueLabel: UILabel!
+    @IBOutlet weak var dateLabel: UILabel!
+    @IBOutlet weak var suggestedTableView: UITableView!
+    @IBOutlet weak var concertInfoView: UIView!
+    @IBOutlet weak var likesCountLabel: UILabel!
+    @IBOutlet weak var dislikesCountLabel: UILabel!
+    @IBOutlet weak var flagButton: UIButton!
+    @IBOutlet weak var likeButton: UIButton!
+    @IBOutlet weak var dislikeButton: UIButton!
+    
+
+    //Variable to store video frame
+    var jamJarContainerFrameInPortrait: CGRect!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view, typically from a nib.
+        
+        //Save the potrait jamjar frame
+        jamJarContainerFrameInPortrait = self.jamJarContainerView.frame
+        
+        //Add action for concert tapped
+        let concertTap = UITapGestureRecognizer(target: self, action: #selector(VideoPageViewController.concertTapped(_:)))
+        self.concertInfoView.addGestureRecognizer(concertTap)
+        
+        // Register the reusable video cell
+        self.suggestedTableView.registerNib(UINib(nibName: "JamJarCell", bundle: nil), forCellReuseIdentifier: "JamJarCell")
+        self.suggestedTableView.registerNib(UINib(nibName: "JamJarHeader", bundle: nil), forCellReuseIdentifier: "JamJarHeader")
+        
+        // Set Report Reasons Data
+        self.reasonsToFlag = ["Accuracy", "Inappropriate", "Quality","Report User"];
+        
+        self.suggestedTableView.reloadData()
+    }
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    deinit {
+        //When the video page is dismissed, remove observers from the AVPlayerController
+        removeOberservsInPlayer()
+    }
+    
+    // Record a view
+    func recordView(videoId: Int!) {
```

Comment:
This is totally nitpicky but can we call this "updateViewCount" or something.  The name of this func makes it sound like we're recording a video or something.